src/components/SamlConfiguration/SamlProviderConfigForm.jsx

```diff
@@ -0,0 +1,359 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import { ValidationFormGroup, Input, StatefulButton, Icon, Button } from '@edx/paragon';
+import StatusAlert from '../StatusAlert';
+
+export const REQUIRED_CONFIG_FIELDS = [
+  'entityId',
+  'metadataSource',
+];
+
+const SUBMIT_STATES = {
+  ERROR: 'error',
+  DEFAULT: 'default',
+  COMPLETE: 'complete',
+  PENDING: 'pending',
+};
+
+
+class SamlProviderConfigForm extends React.Component {
+  state = {
+    invalidFields: {},
+    submitState: SUBMIT_STATES.DEFAULT,
+    enabled: this.props.config ? this.props.config.enabled : false,
+    syncLearnerProfileData: this.props.config &&
```

Comment:
is that an actual format `this.props.config?.syncLearnerProfileData` or a typo? If that's an actual format that's awesome! I mostly just wanted to avoid errors caused by calling an attribute of an object (props.config) that didn't exist.

Reply:
It's a feature in the JS spec called optional chaining specifically for this purpose: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining. I recently learned about it from doing a PR review for the Markhors team and have started using it as well!

Reply:
Neat.